src/TEvent.h

```diff
@@ -35,6 +35,7 @@
 #define ARGUMENT_TYPE_STRING 1
 #define ARGUMENT_TYPE_BOOLEAN 2
 #define ARGUMENT_TYPE_NIL 3
+#define ARGUMENT_TYPE_TABLE 4
```

Comment:
So I can do this now:

```lua
raiseEvent("fffffff", {function() print"hi" end})
````

But not:

```lua
raiseEvent("fffffff", function() print"hi" end)
```

The difference seems kinda silly - mind enabling functions as well?

Reply:
Sure, its trivial to implement. Can you see any reason to differentiate here between tables and functions? They'll both just be references to the registry, and will get treated exactly the same.

Reply:
That is starting to look like a list of `LUA_TXXXX` entries - would it be worth (for consistency) switching to use the standard constants from the Lua headers?

Also it might be that the receiving event will want to know what it receives in each argument - is it possible to detect what was sent as an argument in the event handler or isn't that an issue?

Reply:
> would it be worth (for consistency) switching to use the standard constants from the Lua headers?

Yeah sounds good, in a separate PR to clean that part up!

> Also it might be that the receiving event will want to know what it receives in each argument - is it possible to detect what was sent as an argument in the event handler or isn't that an issue?

I am not sure what do you mean, could you expand on this?

Reply:
What you are saying sounds like it would be hard to differentiate in an event handler whether a particular argument received by that handler is a _table_ or a _function_ if we do not broadcast that information as part of the event details, and if that _was_ the case - __would that be an issue for the handler__?

Reply:
If you're talking about checking the argument type *inside* of Lua, then it should be as simple as calling `type(arg0)` and getting `"table"` or `"function"`

Reply:
See, I keep telling people I am not that hot on Lua...! 🙂